app/models/stacks_file.rb

```diff
@@ -15,12 +15,6 @@ def file_exist?
     path && File.exist?(path)
   end
 
-  def valid?
-    # FIXME: image_exist? is defined in app/models/concerns/djatoka_adapter.rb,
-    #  which is included in StacksImage, but is probably undefined here
-    image_exist?
-  end
```

Comment:
Oh, totally agree.  Just using the opportunity to have a discussion about code. 😎

Some things to consider in a base class like this would be to set a default `true` or `false`.  Depending on the use of your `valid?` method in the app this could have potentially dangerous results.

Another thing I've seen which I like is to `raise NotImplementedError`.  That way if I subclass this, and the `valid?` method gets called anywhere in my routable code path (say I copy the IIIFController for some reason) then a nice big error is raised and you'll see it or it'll be caught by the exception handler.